application/lib/ui/sign_up_screen_ui.dart

```diff
@@ -0,0 +1,111 @@
+
+class SignUpScreen extends StatefulWidget {
+  const SignUpScreen({Key? key}) : super(key: key);
+
+  @override
+  State<SignUpScreen> createState() => _SignUpScreenState();
+}
+
+class _SignUpScreenState extends State<SignUpScreen> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  bool _radioValue = true;
+  final viewModel = SignUpViewModel();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: MyColors.lightBlueColor,
+        title: Text(
+          signUp,
+          style: Styles.textStyles,
+        ),
+      ),
+      body: SingleChildScrollView(
+        child: Container(
+          padding: const EdgeInsets.all(30.0),
+          child: Column(
+            children: [
+              Form(
+                key: _formKey,
+                child: Column(
+                  children: [
+                    SignBox(
+                      const Icon(
+                        Icons.account_circle,
+                      ),
+                      fullName,
+                    ),
+                    SignBox(
+                      const Icon(
+                        Icons.account_circle,
+                      ),
+                      userName,
+                    ),
+                    SignBox(
+                      const Icon(
+                        Icons.mail,
+                      ),
+                      email,
+                    ),
+                    SignBox(
+                      const Icon(
+                        Icons.vpn_key,
+                      ),
+                      password,
+                    ),
+                    Text(
+                      eventsChechBox,
+                      style: Styles.textStyles,
+                    ),
+                    RadioListTile(
+                      title: const Text('Igen'),
+                      value: true,
+                      groupValue: _radioValue,
+                      onChanged: (value) {
+                        setState(() {
+                          _radioValue = value!;
+                        });
+                      },
+                    ),
+                    RadioListTile(
+                      title: const Text('Nem'),
+                      value: false,
+                      groupValue: _radioValue,
+                      onChanged: (value) {
+                        setState(
+                          () {
+                            _radioValue = value!;
+                          },
+                        );
+                      },
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.all(5),
+                      child: MyButton(logIn, () {
+                        if (_formKey.currentState!.validate()) {
+                          if (viewModel.navigateToAdminHome) {
+                            Navigator.pushNamed(context, '/admin_home');
```

Comment:
ezt az if validációt ki tudjuk emelni a ui-hoz tartozó viewModelbe, ahogy a navigáció a start_screen esetében is ki van emelve. Kérlek ezt tegyük meg. A navigáció bemenő paramétere egy bool lesz, nagyjából így: 
`void NavigateToSpecificHomeScreen(bool isAdminUser) {
    if (isAdminUser) {
        Navigator.pushNamed(context, adminHomeScreen);
    } else {
        Navigator.pushNamed(context, userHomeScreen);
    }
}`